### ISSUE TITLE
How to perform set-up ontologies?

### ISSUE THREAD

**Issue by cornellyujy** [UofS-Pulse-Binfo/analyzedphenotypes#130]
Hi @laceysanderson , if I used you example files to make the module run, how to set up its ontologies. Honestly, I have no idea for the set-up ontologies. Could you give me a example for setting up. Thanks~

**Comment by laceysanderson** [comment]
Hi @cornellyujy, did you see these docs: https://analyzedphenotypes.readthedocs.io/en/latest/admin_guide/setup.html? On KnowPulse we use the recommended method (#2 in those docs). The example shown in those docs is for a fictitious "Tripalus databasica" organism which will work with all the other examples in the documentation :-)

**Comment by cornellyujy** [comment]
Yep. I fellow the steps: 1#I added a new CV, and it's just a CV name and description. May I need to add 3 CVs incliding Trait, Method, and Unit like Tripalus?
2# I have added a database, and just a name, and should I follow 1# step to build 3 databases?
3# Our species did not have crop ontology in cropontology, so I gave up this step
4#I added the 3 trait vocabularys that built in step1, the associated database and crop ontology, but these were all empty and just a name.
I am not sure if it make sense?

**Comment by laceysanderson** [comment]
I'm not sure I quite follow so just let me know if I misunderstood you :-)

So you did follow the steps but weren't sure if you needed 3 CVs with matching DBs or only one? You do need 3 unique CVs with their own DB.

The crop ontology is not required. It can also be any ontology you want to associate with your traits :-) 

Yes, it's OK if these CVs are empty. By default, there is a checkbox selected in that same configuration form that allows new terms to be added to those CV/DBs. The way this works is when you upload a file with data, the importer will check to see if the trait, method, unit from the file already exists. If they do then it will just fill them in for you and if they don't you will be asked to fill in more details and then the importer will create them for you with the proper relationships.

**Comment by cornellyujy** [comment]
Yes, you catch that. Thank you so much. :-)
And for the 3 CVs, what's the mean "with their own DB". each CV has one DB or the 3 CVs have one DB? Thanks~
For uploading phenotypic data, 
![image](https://user-images.githubusercontent.com/59180297/161148228-02f95090-d66d-4e50-a6e4-41d20c275ac6.png)
And I want to add "Research Experiment", but I did not find this tripal content under "Toolbar > Content > Tripal Content > Add Tripal Content". So Just guess, maybe I missed some setup, when install this module.  Thanks~

**Comment by laceysanderson** [comment]
> And for the 3 CVs, what's the mean "with their own DB". each CV has one DB or the 3 CVs have one DB?

Each CV has one DB that matches it.

> And I want to add "Research Experiment", but I did not find this tripal content under "Toolbar > Content > Tripal Content > Add Tripal Content".

Oh! That may be a good catch. We apparently use a non-standard content type here but did not explain setting that up... I think you should be able to use the "Project" content type instead without any additional work on your part 🤞

**Comment by cornellyujy** [comment]
Good! Thanks @laceysanderson ,  I got it. 
![image](https://user-images.githubusercontent.com/59180297/161171354-6ebfe6b5-34ab-43d0-a97b-1a20d85a964a.png)
![image](https://user-images.githubusercontent.com/59180297/161171078-76debe8c-ab0a-4772-9308-1e20816d7c40.png)
We have no trait ontology in crop ontology, so may I leave "associated database" and "crop ontology" blank or keep default? Thanks~

**Comment by laceysanderson** [comment]
The database does need to be filled -just with an empty DB (for example "Cucurbita Phenotypic Terms" although the name is not important). Apparently, I made a small mistake above -there is only one database that is used for all three CVs.

**Comment by cornellyujy** [comment]
Thanks Lacey. No worries. I have built the two types. 
Besides, I want to upload the trait data into tables with your test data: SHORT-TestData-1trait2methods.tsv.
But got this error on webpage:
![image](https://user-images.githubusercontent.com/59180297/161558802-fa01ca13-c189-4d3d-a2af-a7e9407662be.png)
displayed in terminal:
![image](https://user-images.githubusercontent.com/59180297/161558935-061c92e0-d553-4e8d-868d-7e72db9d3b23.png)
I have uploaded the genotype data before, and should I fix your test data with the same germplasm in genotype data. Thanks ~

**Comment by laceysanderson** [comment]
This error `grep: : No such file or directory` implies that it couldn't find the file to be imported. If you used the server path option for the file in the importer, please check that it was the full path to the file on your server. I recommend using the file upload option and remembering to click the upload button after selecting the file and before submitting the form.

For the germplasm it will need to already exist so yes, you could change it to match the germplasm from the genotypic data or you could create these 15 germplasm through the create tripal content forms.
